src/input.cpp

```diff
@@ -0,0 +1,397 @@
+    lua_pushinteger(L, nMajorRevision);
+    lua_setfield(L, -2, "major");
+    lua_pushinteger(L, nMinorRevision);
+    lua_setfield(L, -2, "minor");
+    return 1;
+}
+
+// uint32 GetRemotePlaySessionID( InputHandle_t inputHandle );
+EXTERN int luasteam_getRemotePlaySessionID(lua_State *L) {
+    uint64 inputHandle = luasteam::checkuint64(L, 1);
+    lua_pushinteger(L, SteamInput()->GetRemotePlaySessionID(inputHandle));
+    return 1;
+}
+
+namespace luasteam {
+
+void add_input(lua_State *L) {
+    lua_createtable(L, 0, 31);
+    add_func(L, "activateActionSet", luasteam_activateActionSet);
+    add_func(L, "activateActionSetLayer", luasteam_activateActionSetLayer);
+    add_func(L, "deactivateActionSetLayer", luasteam_deactivateActionSetLayer);
+    add_func(L, "deactivateAllActionSetLayers", luasteam_deactivateAllActionSetLayers);
+    add_func(L, "getActiveActionSetLayers", luasteam_getActiveActionSetLayers);
+    add_func(L, "getActionSetHandle", luasteam_getActionSetHandle);
+    add_func(L, "getAnalogActionData", luasteam_getAnalogActionData);
+    add_func(L, "getAnalogActionHandle", luasteam_getAnalogActionHandle);
+    add_func(L, "getAnalogActionOrigins", luasteam_getAnalogActionOrigins);
+    add_func(L, "getConnectedControllers", luasteam_getConnectedControllers);
+    add_func(L, "getControllerForGamepadIndex", luasteam_getControllerForGamepadIndex);
+    add_func(L, "getCurrentActionSet", luasteam_getCurrentActionSet);
+    add_func(L, "getDigitalActionData", luasteam_getDigitalActionData);
+    add_func(L, "getDigitalActionHandle", luasteam_getDigitalActionHandle);
+    add_func(L, "getDigitalActionOrigins", luasteam_getDigitalActionOrigins);
+    add_func(L, "getGamepadIndexForController", luasteam_getGamepadIndexForController);
+    add_func(L, "getGlyphForActionOrigin", luasteam_getGlyphForActionOrigin);
+    add_func(L, "getInputTypeForHandle", luasteam_getInputTypeForHandle);
+    add_func(L, "getMotionData", luasteam_getMotionData);
+    add_func(L, "getStringForActionOrigin", luasteam_getStringForActionOrigin);
+    add_func(L, "runFrame", luasteam_runFrame);
+    add_func(L, "setLEDColor", luasteam_setLEDColor);
+    add_func(L, "showBindingPanel", luasteam_showBindingPanel);
+    add_func(L, "stopAnalogActionMomentum", luasteam_stopAnalogActionMomentum);
+    add_func(L, "triggerHapticPulse", luasteam_triggerHapticPulse);
+    add_func(L, "triggerRepeatedHapticPulse", luasteam_triggerRepeatedHapticPulse);
+    add_func(L, "triggerVibration", luasteam_triggerVibration);
+    add_func(L, "getActionOriginFromXboxOrigin", luasteam_getActionOriginFromXboxOrigin);
+    add_func(L, "translateActionOrigin", luasteam_translateActionOrigin);
+    add_func(L, "getDeviceBindingRevision", luasteam_getDeviceBindingRevision);
+    add_func(L, "getRemotePlaySessionID", luasteam_getRemotePlaySessionID);
+    lua_setfield(L, -2, "input");
+}
+
+void init_input(lua_State *L) {
+    SteamInput()->Init();
```

Comment:
I don't think we should do that here. It should have an `init` (and shutdown) function that people can call, so people that don't use input stuff don't pay for it.

This function is mostly for things like creating CallbackListeners, which we don't need here.